centreon-certified/bsm/bsm_connector-apiv2.lua

```diff
@@ -0,0 +1,300 @@
+
+  -- apply users params and check syntax of standard ones
+  self.sc_params:param_override(params)
+  self.sc_params:check_params()
+
+  self.sc_macros = sc_macros.new(self.sc_params.params, self.sc_logger)
+  self.format_template = self.sc_params:load_event_format_file(true)
+  self.sc_params:build_accepted_elements_info()
+  self.sc_flush = sc_flush.new(self.sc_params.params, self.sc_logger)
+
+  local categories = self.sc_params.params.bbdo.categories
+  local elements = self.sc_params.params.bbdo.elements
+
+  self.format_event = {
+    [categories.neb.id] = {
+      [elements.host_status.id] = function () return self:format_event_host() end,
+      [elements.service_status.id] = function () return self:format_event_service() end
+    },
+    [categories.bam.id] = {}
+  }
+  self.send_data_method = {
+    [1] = function (data, element) return self:send_data(data, element) end
+  }
+  
+  -- return EventQueue object
+  setmetatable(self, { __index = EventQueue })
+  return self
+end
+
+-- Format XML file with host infoamtion
+function EventQueue:format_event_host()
+  local xml_host_severity = "<host_severity>" .. self.sc_common:ifnil_or_empty(self.sc_broker:get_severity(self.sc_event.event.host_id) , '0') .. "</host_severity>"
+  local xml_url = self.sc_common:ifnil_or_empty(self.sc_event.event.cache.service.action_url, 'no action url for this host')
+  local xml_notes = "<host_notes>" .. self.sc_common:ifnil_or_empty(self.sc_event.event.cache.service.notes, 'OS not set') .. "</host_notes>"
+  
+  self.sc_event.event.formated_event = {
+    "<event_data>"
+      .. "<hostname>" .. hostname .. "</hostname>"
+      .. xml_host_severity .. xml_notes
+      .. "<url>" .. xml_url .. "</url>"
+      .. "<source_ci>" .. ifnil_or_empty(self.source_ci, 'Centreon') .. "</source_ci>"
+      .. "<source_host_id>" .. ifnil_or_empty(self.sc_event.event.host_id, '0') .. "</source_host_id>"
+      .. "<scheduled_downtime_depth>" .. ifnil_or_empty(self.sc_event.event.scheduled_downtime_depth, '0') .. "</scheduled_downtime_depth>"
+      .. "</event_data>"
+  }
+end
+
+-- Format XML file with service infoamtion
+function EventQueue:format_event_service()
+  local xml_url = self.sc_common:ifnil_or_empty(self.sc_event.event.cache.service.notes_url, 'no notes url for this service')
+  local xml_service_severity = "<service_severity>" .. self.sc_common:ifnil_or_empty(self.sc_broker:get_severity(self.sc_event.event.host_id, self.sc_event.event.service_id) , '0') .. "</service_severity>"
+    
+  self.sc_event.event.formated_event = {
+      "<event_data>"
+        .. "<hostname>" .. hostname .. "</hostname>"
+        .. "<svc_desc>" .. service_description .. "</svc_desc>"
+        .. "<state>" ..self.sc_event.event.state .. "</state>"
+        .. "<last_update>" ..self.sc_event.event.last_update .. "</last_update>"
+        .. "<output>" .. string.match(e.output, "^(.*)\n") .. "</output>"
+        .. xml_service_severity
```

Comment:
same comment about code, writing xml and randomly putting a variable in the middle of it